### PR TITLE
Add --deprecation-days-XXX flag for breaking-action

### PR DESCRIPTION
**Pull request by christopher-gerdes**
Hey everyone! 

Right now it is not possible to change the default parameters for the sunsetting periods for deprecation of beta and stable endpoints. This PR adds those parameters and takes over explicitly the default parameters used in oasdiff.

Would be great to have this capability and I am happy to change the PR if anything is missing!

### CODE REVIEW

File: breaking/action.yml [hunk 1]
```diff
@@ -18,6 +18,14 @@ inputs:
...
     required: false
     default: 'false'
+  deprecation-days-beta:
+    description: 'Consider minimum sunset period for deprecation of beta API endpoints'
+    required: false
+    default: '31'
+  deprecation-days-stable:
+    description: 'Consider minimum sunset period for deprecation of stable API endpoints'
+    required: false
+    default: '180'
```
Comment (effoeffi) [inline breaking/action.yml:28]: While the script checks if `deprecation_days_beta` and `deprecation-days-stable` are not empty, assigning a default value implies that these variables will always have a value, right?

**Review by effoeffi — COMMENTED** [review]
Please, add a test. I suggest adding a new yaml with deprecated endpoint.

File: breaking/action.yml (hunk 1 above)
Comment (christopher-gerdes) [reply breaking/action.yml]: That's right. I removed the default values to fallback to the oasdiff defaults. Kept them in the README though 👍

**Comment by christopher-gerdes** [comment]
> Please, add a test. I suggest adding a new yaml with deprecated endpoint.

Added a test. Bit tricky with the moving date situation. I guess I would either need to dynamically create the revision file or change the date of the container that oasdiff is performing on. I chose the first option but happy to reiterate if you have a better idea.

**Comment by effoeffi** [comment]
> Added a test. Bit tricky with the moving date situation. I guess I would either need to dynamically create the revision file or change the date of the container that oasdiff is performing on. I chose the first option but happy to reiterate if you have a better idea.

What do you think about adding a static OpenAPI file `base-deprecation.yaml`? This file would contain a placeholder instead of a sunset date. During the build, this placeholder would be replaced with the actual sunset date using a tool like sed or echo?

For example:
```
/pets:
 get:
  deprecated: true
  x-sunset: {{SUNSET_DATE}}
```

**Comment by christopher-gerdes** [comment]
> > Added a test. Bit tricky with the moving date situation. I guess I would either need to dynamically create the revision file or change the date of the container that oasdiff is performing on. I chose the first option but happy to reiterate if you have a better idea.
> 
> What do you think about adding a static OpenAPI file `base-deprecation.yaml`? This file would contain a placeholder instead of a sunset date. During the build, this placeholder would be replaced with the actual sunset date using a tool like sed or echo?
> 
> For example:
> 
> ```
> /pets:
>  get:
>   deprecated: true
>   x-sunset: {{SUNSET_DATE}}
> ```

Definitely makes the tests more readable. Did that 👍

**Review by effoeffi — COMMENTED** [review]
Looks like the test failed. Can you please take a look?
https://github.com/oasdiff/oasdiff-action/actions/runs/6927590283/job/18902620085

**Comment by christopher-gerdes** [comment]
> Looks like the test failed. Can you please take a look? https://github.com/oasdiff/oasdiff-action/actions/runs/6927590283/job/18902620085

Looks like a line of gibberish slipped into the yaml specification. Sorry, for letting this linger so long here. Should be good now.

**Review by effoeffi — APPROVED** [review]
Thanks :)
Please, join our slack community https://join.slack.com/t/oasdiff/shared_invite/zt-27l6y20xw-hcwzpcA7v8FobfRogTCgtg

**Pull request merged**